temporalcli/internal/printer/printer.go

```diff
@@ -50,6 +54,42 @@ func (p *Printer) Printlnf(s string, v ...any) {
 	p.Println(fmt.Sprintf(s, v...))
 }
 
+// When called for JSON with indent, this will create an initial bracket and
+// make sure all [Printer.PrintStructured] calls get commas properly to appear
+// as a list (but the indention and multiline posture of the JSON remains). When
+// called for JSON without indent, this will make sure all
+// [Printer.PrintStructured] is on its own line (i.e. JSONL mode). When called
+// for non-JSON, this is a no-op.
```

Comment:
This (and the if checks below) says to me that rather than using an empty `JSONIndent` as a special indication to do JSONL, we should instead just have the `JSON` field be `JSON | JSONL` and just be explicit about what mode we're in.

Reply:
> we should instead just have the `JSON` field be `JSON | JSONL`

More like `JSON | JSONL | Neither`, but it's ugly in Go. I considered this (and I considered two booleans), but `JSONIndent` is how Go does it (where empty string means no multiline, ref https://pkg.go.dev/encoding/json#Encoder.SetIndent). Throughout the Go JSON printing code they check whether indentation is set. And this option gives the caller the freedom to set the indent they want, even though we're the caller. However, if the way Go did this is too confusing, I can make this a Go enum and add a `isJSON()` that checks for those two values.

Reply:
I'd prefer that I think but it's not a big deal. I'll approve and leave it up to you. Lack up sum types :cry: